massa-execution-worker/src/execution.rs

```diff
@@ -1341,4 +1356,54 @@ impl ExecutionState {
     pub fn get_address_future_deferred_credits(&self, address: &Address) -> BTreeMap<Slot, Amount> {
         context_guard!(self).get_address_future_deferred_credits(address, self.config.thread_count)
     }
+
+    /// Initialize the hashmap of addresses from the vesting file
+    fn init_vesting_registry(
+        config: &ExecutionConfig,
+    ) -> Result<PreHashMap<Address, Vec<VestingRange>>, ExecutionError> {
+        let mut hashmap: PreHashMap<Address, Vec<VestingRange>> = serde_json::from_str(
+            &std::fs::read_to_string(&config.initial_vesting_path).map_err(|err| {
+                ExecutionError::InitVestingError(format!(
+                    "error loading initial vesting file  {}",
+                    err
+                ))
+            })?,
+        )
+        .map_err(|err| {
+            ExecutionError::InitVestingError(format!(
+                "error on deserialize initial vesting file  {}",
+                err
+            ))
+        })?;
+
+        for v in hashmap.values_mut() {
+            let mut current_index = 1;
+            let temp_list = v.clone();
```

Comment:
Instead of making a clone and index fetching for the next could you use `.next()` method of an element ?

Reply:
i don't think so, next() need a mutable reference on the same iter.

Fox example this do not compile because iter_vesting.next() take the ownership on iter_vesting. 

```rust
.map(|mut res: PreHashMap<Address, Vec<VestingRange>>| {
            let _ = res.values_mut().map(|value| {
                let mut iter_vesting = value.iter_mut();

                for vesting_slot in iter_vesting {
                    if let Some(next) = iter_vesting.next() {
                        if let Ok(prev_slot) = next.start_slot
                            .get_prev_slot(config.thread_count) {
                            vesting_slot.end_slot = prev_slot;
                        } else {
                            error!("error on get prev slot for init vesting : {}", next.start_slot)
                        }
                    } else {
                        vesting_slot.end_slot = Slot {
                            thread: config.thread_count,
                            period: u64::MAX,
                        };
                    }
                }
            });
            res
        })

```

Reply:
I proposed a refactor here : https://github.com/massalabs/massa/pull/3549/commits/aecfe3ae70c795afcee948d0b74e1c4442df834e